src/stm32/spi.c

```diff
@@ -21,7 +21,7 @@ DECL_ENUMERATION("spi_bus", "spi1", 1);
 DECL_CONSTANT_STR("BUS_PINS_spi1", "PA6,PA7,PA5");
 DECL_ENUMERATION("spi_bus", "spi1a", 2);
 DECL_CONSTANT_STR("BUS_PINS_spi1a", "PB4,PB5,PB3");
-#if CONFIG_MACH_STM32F0 || CONFIG_MACH_STM32F2 || CONFIG_MACH_STM32F4
+#if !CONFIG_MACH_STM32F1
```

Comment:
Couldn't you cover all processors, new and current with this instead:
#ifdef SPI2

Reply:
Alas, no.  The stm32f1 has SPI2, it just doesn't have ability to route it to pins PC2,PC3,PB10.

-Kevin

Reply:
Ah, you're right.  Although, I have to wonder if it really needs to be switched out. In the same manner that another grouping of SPI pins may not map out on a certain board yet are enabled, if someone setting up a STM32F1 board never enables them because they aren't mapped in the schematic or can't be used, would it be considered the same thing?

Reply:
The stm32f1 has a goofy alternative mapping system, so I don't really know what would happen if a user tried to enable spi2a on it.  (In contrast, on the other stm32 chips, if a user enabled a bus that didn't exist, it just wouldn't work - but should not cause internal conflicts.)  I don't think the #ifdef is super critical, but I don't see a harm in it either.

-Kevin